packages/flutter_tools/lib/src/asset.dart

```diff
@@ -621,39 +625,49 @@ class ManifestAssetBundle implements AssetBundle {
     String assetBasePath,
     List<Uri> wildcardDirectories,
     Directory projectDirectory, {
-    List<String> excludeDirs = const <String>[],
+    String? flavor,
   }) {
     final List<DeferredComponent>? components = flutterManifest.deferredComponents;
     final Map<String, Map<_Asset, List<_Asset>>> deferredComponentsAssetVariants = <String, Map<_Asset, List<_Asset>>>{};
     if (components == null) {
       return deferredComponentsAssetVariants;
     }
     for (final DeferredComponent component in components) {
-      deferredComponentsAssetVariants[component.name] = <_Asset, List<_Asset>>{};
       final _AssetDirectoryCache cache = _AssetDirectoryCache(_fileSystem);
-      for (final Uri assetUri in component.assets) {
-        if (assetUri.path.endsWith('/')) {
-          wildcardDirectories.add(assetUri);
+      final Map<_Asset, List<_Asset>> componentAssets = <_Asset, List<_Asset>>{};
+      for (final AssetsEntry assetsEntry in component.assets) {
+        if (assetsEntry.uri.path.endsWith('/')) {
+          wildcardDirectories.add(assetsEntry.uri);
           _parseAssetsFromFolder(
             packageConfig,
             flutterManifest,
             assetBasePath,
             cache,
-            deferredComponentsAssetVariants[component.name]!,
-            assetUri,
+            componentAssets,
+            assetsEntry.uri,
           );
         } else {
           _parseAssetFromFile(
             packageConfig,
             flutterManifest,
             assetBasePath,
             cache,
-            deferredComponentsAssetVariants[component.name]!,
-            assetUri,
-            excludeDirs: excludeDirs,
+            componentAssets,
+            assetsEntry.uri,
           );
         }
       }
+
+      componentAssets.removeWhere((_Asset asset, List<_Asset> variants) {
+        if (!asset.matchesFlavor(flavor)) {
+          _logger.printTrace('Skipping assets entry "${asset.entryUri.path}" since '
+            'its configured flavor(s) did not match the provided flavor (if any).\n'
+            'Configured flavors: ${asset.flavors.join(', ')}\n');
```

Comment:
do we need this?

Reply:
I rely on this to see if an asset got bundled by `flutter build` in integration tests ([code](https://github.com/flutter/flutter/pull/132985/files#diff-53383b32b975bfed6875306dfb98911cad077a5251ca0591c5b0e125fb4a0f05R49)). I am not a big fan of depending on logs for testing, but I think the other way to do this would be to delve into the output of `flutter build` and look for asset files there, which might require a significant amount of effort to implement 🤷 .

Reply:
this will get written to AssetManifest.json, right?

Reply:
Yeah, one could decode the AssetManifest and effectively get a list of assets that were bundled

Reply:
ok, let me try to modify this test to add that

Reply:
> ok, let me try to modify this test to add that

@christopherfujino IIRC, you have a branch that does this now. Feel free to push merge it into this branch or send a link to it, and I can integrate it

Reply:
thanks for the ping, opened https://github.com/andrewkolos/flutter/pull/1

Reply:
didn't run it on CI, but it passes for me locally